eos-updater/poll-common.c

```diff
@@ -407,20 +380,77 @@ cleanstr (gchar *s)
   return s;
 }
 
-EosMetricsInfo *
-eos_metrics_info_new (const gchar *booted_ref)
+static GVariant *
+checkpoint_blocked_event_payload_new (const char         *booted_ref,
+                                      const char         *target_ref,
+                                      EuuCheckpointBlock  reason)
 {
   g_autoptr(GHashTable) hw_descriptors = NULL;
-  g_autoptr(EosMetricsInfo) info = NULL;
+  g_autofree gchar *vendor, *product;
+  const char *reason_nick;
 
   hw_descriptors = get_hw_descriptors ();
 
-  info = g_object_new (EOS_TYPE_METRICS_INFO, NULL);
-  info->vendor = cleanstr (g_strdup (g_hash_table_lookup (hw_descriptors, VENDOR_KEY)));
-  info->product = cleanstr (g_strdup (g_hash_table_lookup (hw_descriptors, PRODUCT_KEY)));
-  info->ref = g_strdup (booted_ref);
+  vendor = cleanstr (g_strdup (g_hash_table_lookup (hw_descriptors, VENDOR_KEY)));
+  product = cleanstr (g_strdup (g_hash_table_lookup (hw_descriptors, PRODUCT_KEY)));
+
+  reason_nick = euu_checkpoint_block_to_string (reason);
+
+  return g_variant_new ("(sssss)", /* snake noises are valid type signatures */
+                        vendor,
+                        product,
+                        booted_ref,
+                        target_ref,
+                        reason_nick ?: "");
+}
+
+static gboolean
+stop_aggregate_timer (gpointer data)
+{
+  g_return_val_if_fail (EMTR_IS_AGGREGATE_TIMER (data), G_SOURCE_REMOVE);
```

Comment:
Maybe the compiler will elide this function, but I'd assume this function needs to be under `
#ifdef HAS_EOSMETRICS_0` for the `emtr` symbols and macros.

Reply:
Good point!

The other option is to not wait 1 second and just stop it immediately after it starts. The aggregate will still get reported even if its count is 0…

Reply:
I moved the `#ifdef` out to encompass both functions, and added an `#else` branch that defines a stub `maybe_send_checkpoint_blocked_event()`.